Rewrite "Start Tasks"; add illustrations

mikenikles opened the pull request:
Close #617.

<a href="https://gitpod.io/#https://github.com/gitpod-io/website/pull/621"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>

mikenikles commented:
Hey @rfay,

I rewrote the Start Tasks documentation based on [your comment](https://github.com/gitpod-io/gitpod/issues/4205#issuecomment-858852529).

I'd love your feedback and whether this PR helps explain this topic better. The preview URL is: https://deploy-preview-621--gitpod-kumquat.netlify.app/docs/config-start-tasks

Thanks, Mike

rfay commented on src/routes/docs/config-start-tasks.md:
```diff
@@ -9,83 +9,66 @@ title: Start Tasks
+In this startup scenario, you can see how Prebuilds impact the execution order of tasks:
 
-Task properties will control when a command is executed. The table below provides an overview of the different starting scenarios.
+![Start tasks for Prebuilds & New Workspace](../../images/docs/beta/configure/start-tasks/prebuilds-new-workspace.png)
```
The relative URL doesn't seem to work here. Would be nice for reviewing.  See https://github.com/gitpod-io/website/blob/a968e7adc146dce0580df1f54f691e62f7881f68/src/routes/docs/config-start-tasks.md

rfay commented on src/routes/docs/config-start-tasks.md:
```diff
@@ -94,20 +77,57 @@ tasks:
   - openMode: split-right
-    command: echo SplitTerminal
-  - openIn: left
-    command: echo LeftPanelTerminal
+    command: sh ./scripts/start-db.sh
+    name: DB Server
```
This would be better with the command or the name as the lead thing. openMode is really an attribute and it's confusing to see it as the first one.

rfay commented on src/routes/docs/config-start-tasks.md:
````diff
@@ -94,20 +77,57 @@ tasks:
+1. The second terminal only has a `command` task which starts the dev server. The application code connects to the database that is started in the first terminal (e.g. via localhost:3306).
+
+```
+tasks:
+  - init: |
+      npm install
+      npm run configure-database
+    command: npm run start-database
+  - command: npm run dev
````
Just a note about something that I'm sure can't be fiddled with at this point... Bundling both "init" and "command" under a single task is odd.  It would have been better for them to be separate tasks. But I know that's out of scope. But it has contributed to me stumbling around in this area. 

I think your examples would be better if they mostly led with a name so it was clear that each task can have both init and command.

```suggestion
tasks:
  - name: "npm setup"
     init: |
      npm install
      npm run configure-database
    command: npm run start-database
  - name: "npm run dev"
    command: npm run dev
```

rfay reviewed (changes requested):
The images were broken, so couldn't see them on https://github.com/gitpod-io/website/blob/a968e7adc146dce0580df1f54f691e62f7881f68/src/routes/docs/config-start-tasks.md but overall this is really helpful, thanks for all the great work!

rfay commented:
Thanks, I can see all the illustrations now. 

I think the several illustrations where "with prebuild" and "without prebuild" are the same would be clearer if they just said "with or without prebuild" and had just one column.

mikenikles commented:
Excellent feedback, thank you so much Randy! I merged it and will see what the feedback looks like from the wider community.

shaal commented:
New documentation is 🔥 

A few thoughts -
`name` in start-tasks is optional. Is that correct?

I think it will be very useful to add examples, like running `make` or `composer install`.
What are real scenarios where we should use `init` vs `command` vs `before`

I remember there was an example that show how to get 1 terminal to wait for the other terminal to finish something. Perhaps it's good to add it to that doc?

mikenikles commented:
Thanks for the feedback 🙏!

Yes, `name` is optional. I considered adding a "Required" field to each property at https://www.gitpod.io/docs/references/gitpod-yml, but it turns out pretty much every property is optional.

In terms of more examples, I'll have to think about that and see what makes sense to add. The new one I added as part of #651  doesn't have a `before`, but is taken from a community member's project.

Excellent advice, I added a `gp sync-await` example in #651.
